README.md

```diff
@@ -5,28 +5,45 @@
 
 > [!CAUTION]
-> This library is still under rapid development and is very much a work in progress!
+> This library is still under rapid development and is a work in progress!
 >
-> Some key features are not yet implemented, the API will almost certainly change in breaking ways, and we cannot
-> yet guarantee correctness in all cases.
+> Some key features are not yet implemented, both the API and the serialized format are likely to change in breaking ways,
+> and we cannot yet guarantee correctness in all cases.
 
-The major components of Vortex are (will be!):
+The major components of Vortex are:
 
 * **Logical Types** - a schema definition that makes no assertions about physical layout.
-* **Encodings** - a pluggable set of physical layouts. Vortex ships with several state-of-the-art lightweight
-  compression codecs that have the potential to support GPU decompression.
-* **Compression** - recursive compression based on stratified samples of the input.
-* **Compute** - basic compute kernels that can operate over compressed data. Note that Vortex does not intend to become
-  a full-fledged compute engine, but rather to provide the ability to implement basic compute operations as may be
-  required for efficient scanning & pushdown operations.
+* **Zero-Copy to Arrow** - "canonicalized" (i.e., fully decompressed) Vortex arrays can be zero-copy converted to/from Apache Arrow arrays.
+* **Extensible Encodings** - a pluggable set of physical layouts. In addition to the builtin set of Arrow-compatible encodings,
+  the Vortex repository includes a number of state-of-the-art encodings (e.g., FastLanes, ALP, FSST, etc.) that are implemented
+  as extensions. While arbitrary encodings can be implemented as extensions, we have intentionally chosen a small set
+  of encodings that are highly data-parallel, which in turn allows for efficient vectorized decoding, random access reads,
+  and (in the future) decompression on GPUs.
+* **Cascading Compression** - data can be recursively compressed with multiple nested encodings.
+* **Pluggable Compression Strategies** - the built-in Compressor is based on BtrBlocks, but other strategies can trivially be used instead.
+* **Compute** - basic compute kernels that can operate over encoded data. Note that Vortex does not intend to become
+  a full-fledged compute engine, but rather to implement lower-level compute operations required for efficient scanning & pushdown.
 * **Statistics** - each array carries around lazily computed summary statistics, optionally populated at read-time.
   These are available to compute kernels as well as to the compressor.
-* **Serde** - zero-copy serialization. Useful as a building block in creating IPC or file formats that contain
-  compressed arrays.
+* **Serialization** - Zero-copy serialization, both for IPC and for file formats.
+* **Vinyl Columnar File Format (in progress)** - A modern file format that uses the Vortex serde library to store compressed array data.
```

Comment:
Do we have plans for a native partitioned file format? Maybe that’s what Layouts can achieve? That’s something I’d be looking for if I came to the repo a year ago wondering how to replace Hail’s custom format for a dataset with 100k partitions.

Reply:
I think spiral will generally be the partitioned file format. We will extend vinyl to be multi file but that will be column wise and not row wise. If you have good pruning and random access why have physical file partitioning?